baselines/fedrep/fedrep/client.py

```diff
@@ -0,0 +1,296 @@
+            model_keys.extend(
+                [
+                    k
+                    for k in self.model_manager.model.state_dict().keys()
+                    if k.startswith("_head")
+                ]
+            )
+
+        state_dict = OrderedDict(
+            (k, torch.from_numpy(v)) for k, v in zip(model_keys, parameters)
+        )
+
+        self.model_manager.model.set_parameters(state_dict)
+
+
+# pylint: disable=E1101
+def get_client_fn_simulation(
+    config: DictConfig, client_state_save_path: str = ""
+) -> Callable[[str], Client]:
+    """Generate the client function that creates the Flower Clients.
+
+    Parameters
+    ----------
+    model : DictConfig
+        The model configuration.
+    cleint_state_save_path : str
+        The path to save the client state.
+
+    Returns
+    -------
+    Tuple[Callable[[str], FlowerClient], DataLoader]
+        A tuple containing the client function that creates Flower Clients and
+        the DataLoader that will be used for testing
+    """
+    assert config.model_name.lower() in [
+        "cnncifar10",
+        "cnncifar100",
+    ], f"Model {config.model_name} not implemented"
+
+    # load dataset (cifar10/cifar100) and clients' data indices
+    try:
+        partition_path = (
+            PROJECT_DIR / "datasets" / config.dataset.name / "partition.pkl"
+        )
+        print(f"Loading partition from {partition_path}")
+        with open(partition_path, "rb") as pickle_file:
+            partition = pickle.load(pickle_file)
+        data_indices: Dict[int, Dict[str, List[int]]] = partition["data_indices"]
+    except FileNotFoundError as error:
+        print(f"Partition not found at {partition_path}")
+        raise error
+
+    # - you can define your own data transformation strategy here -
+    train_data_transform = transforms.Compose(
+        [
+            transforms.RandomCrop(32, padding=4),
+            transforms.RandomHorizontalFlip(),
+            # transforms.ToTensor(),
+            transforms.Normalize(MEAN[config.dataset.name], STD[config.dataset.name]),
+        ]
+    )
+    test_data_transform = transforms.Compose(
+        [
+            # transforms.ToTensor(),
+            transforms.Normalize(MEAN[config.dataset.name], STD[config.dataset.name])
+        ]
+    )
+    dataset = call_dataset(
+        dataset_name=config.dataset.name,
+        root=PROJECT_DIR / "datasets" / config.dataset.name,
+        train_data_transform=train_data_transform,
+        test_data_transform=test_data_transform,
+    )
+
+    # pylint: disable=E1101
+    def client_fn(cid: str) -> Client:
+        """Create a Flower client representing a single organization."""
+        cid_use = int(cid)
+
+        trainset = Subset(dataset, indices=data_indices[cid_use]["train"])
+        testset = Subset(dataset, indices=data_indices[cid_use]["test"])
+
+        # Create the train loader
+        trainloader = DataLoader(trainset, config.batch_size, shuffle=True)
```

Comment:
I'm seeing some `ValueError` thrown here:
```
  File "/home/javier/projects/flower-karhoutam/baselines/fedrep/fedrep/client.py", line 262, in client_fn
    trainloader = DataLoader(trainset, config.batch_size, shuffle=True)
  File "/home/javier/.cache/pypoetry/virtualenvs/fedrep-qLhlcELz-py3.10/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 350, in __init__
    sampler = RandomSampler(dataset, generator=generator)  # type: ignore[arg-type]
  File "/home/javier/.cache/pypoetry/virtualenvs/fedrep-qLhlcELz-py3.10/lib/python3.10/site-packages/torch/utils/data/sampler.py", line 143, in __init__
    raise ValueError(f"num_samples should be a positive integer value, but got num_samples={self.num_samples}")
ValueError: num_samples should be a positive integer value, but got num_samples=0
```

Reply:
Seems this bug is raised by the 0 length of `trainset.indices` and `DataLoader(shuffle=True)`.  
Nothing wrong on my workspace... 😐
Could you provide the commands for me to reproduce this bug?